src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -388,13 +424,15 @@ object CheckTypes extends Pass {
     }
 
     def bulk_equals(t1: Type, t2: Type, flip1: Orientation, flip2: Orientation): Boolean = {
-      //;println_all(["Inside with t1:" t1 ",t2:" t2 ",f1:" flip1 ",f2:" flip2])
       (t1, t2) match {
         case (ClockType, ClockType) => flip1 == flip2
         case (_: UIntType, _: UIntType) => flip1 == flip2
         case (_: SIntType, _: SIntType) => flip1 == flip2
         case (_: FixedType, _: FixedType) => flip1 == flip2
         case (_: AnalogType, _: AnalogType) => true
+        case (AsyncResetType, AsyncResetType) => flip1 == flip2
```

Comment:
Why is there an asymmetry here between AsyncResetType and ResetType.
Is it intended that ART ART is ok, RT can connect to RT, UInt and ART,
but ART cannot connect to UInt.
Could be fine, but I don't get it.

Reply:
The asymmetry comes from the fact that `ResetType` gets inferred by what's connected to it. Normally you have to cast all types to connect them (eg. must cast from AsyncResetType to UInt if you want to connect), but the types of things driving ResetTypes tells us what the type should be:
```
wire x : UInt<1>
wire w : Reset
w <= x ; This will cause us to infer w to be UInt<1>
```
You can't connect a Reset to a UInt (nor AsyncResetType) without casting because the inference only flows one way.